src/bms/command/channel.rs

```diff
@@ -183,109 +178,142 @@ pub fn read_channel_beat(channel: &str) -> Option<Channel> {
     Some(Channel::Note { kind, side, key })
 }
 
-/// Reads a key from a character. (For PMS)
-fn get_key_pms_bme_type(key: char) -> Option<Key> {
-    use Key::*;
-    Some(match key {
-        '1' => Key1,
-        '2' => Key2,
-        '3' => Key3,
-        '4' => Key4,
-        '5' => Key5,
-        '6' => Key8,
-        '7' => Key9,
-        '8' => Key6,
-        '9' => Key7,
-        _ => return None,
-    })
+/// Key channel family for different key mapping schemes.
+///
+/// Mappings used by each family:
+/// - Beat:
+///   - Chars: '1'..'7','6' scratch, '7' free zone, '8'->Key6, '9'->Key7
+///   - Char -> (PlayerSide, Key):
+///     - '1'..'5' => (Player1, Key1..Key5), '6' => (Player1, Scratch),
+///       '7' => (Player1, FreeZone), '8' => (Player1, Key6), '9' => (Player1, Key7)
+/// - PmsBmeType:
+///   - Beat -> this: Scratch=>Key8, FreeZone=>Key9 (others unchanged)
+///   - This -> Beat: Key8=>Scratch, Key9=>FreeZone (others unchanged)
+/// - Pms:
+///   - Beat -> this: (P2,Key2..Key5) remapped to (P1,Key6..Key9); (P1,Key1..Key5) unchanged
+///   - This -> Beat: Key6..Key9 => (P2,Key2..Key5); Key1..Key5 => (P1,Key1..Key5)
+/// - BeatNanasi:
+///   - Beat -> this: FreeZone=>FootPedal
+///   - This -> Beat: FootPedal=>FreeZone
+/// - DscOctFp:
+///   - Beat -> this: (P2,Key1)=>FootPedal, (P2,Key2..Key7)=>Key8..Key13, (P2,Scratch)=>ScratchExtra; (P1,Key1..Key7|Scratch) unchanged; side becomes P1
+///   - This -> Beat: reverse of above
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+pub enum ModeKeyChannel {
```

Comment:
I think that it's not good about making key mapping as an enum. Because this makes hard to add, modify and extend the mapping.
For maintainability, they should be each own type and implement the common trait like `KeyMapping` that converts the side and key (or from the channel number).